packages/kbn-apm-synthtrace/src/scenarios/spiked_latency.ts

```diff
@@ -0,0 +1,151 @@
+                    'orchestrator.cluster.id': CLUSTER[index].clusterId,
+                    'orchestrator.resource.id': generateShortId(),
+                    'cloud.provider': CLOUD_PROVIDERS[Math.floor(Math.random() * 3)],
+                    'cloud.region': CLOUD_REGION[index],
+                    'cloud.availability_zone': `${CLOUD_REGION[index]}a`,
+                    'cloud.project.id': generateShortId(),
+                    'cloud.instance.id': generateShortId(),
+                    'log.file.path': `/logs/${generateLongId()}/error.txt`,
+                  })
+                  .timestamp(timestamp);
+              });
+          });
+      }
+
+      const serviceInstances = serviceNames.map((serviceName) =>
+        apm
+          .service({ name: serviceName, environment: ENVIRONMENT, agentName: 'go' })
+          .instance(`my-instance`)
+      );
+
+      const transactionNames = [
+        'GET /order',
+        'GET /product',
+        alwaysSpikeTransactionName,
+        sometimesSpikeTransactionName,
+      ];
+
+      const buildTransactions = (serviceInstance: Instance, transactionName: string) => {
+        const interval = random(1, 100, false);
+        const rangeWithInterval = range.interval(`${interval}s`);
+
+        return rangeWithInterval.generator((timestamp, i) => {
+          const duration = getDuration(transactionName);
+          return serviceInstance
+            .transaction({ transactionName })
+            .timestamp(timestamp)
+            .duration(duration)
+            .children(
+              serviceInstance
+                .span({
+                  spanName: 'GET apm-*/_search',
+                  spanType: 'db',
+                  spanSubtype: 'elasticsearch',
+                })
+                .duration(duration * 0.9)
+                .destination('elasticsearch')
+                .timestamp(timestamp)
+                .outcome('success')
+            )
+            .success();
+        });
+      };
+
+      return [
+        withClient(
+          logsEsClient,
+          logger.perf('generating_logs', () =>
+            serviceNames.map((serviceName) => buildLogs(serviceName))
+          )
+        ),
+        withClient(
+          apmEsClient,
+          logger.perf('generating_apm_events', () =>
+            serviceInstances.flatMap((serviceInstance) =>
+              transactionNames.flatMap((transactionName) =>
+                buildTransactions(serviceInstance, transactionName)
+              )
+            )
+          )
+        ),
+      ];
+    },
+  };
+};
+
+function getDuration(transactionName: string) {
+  const spikedDuration = random(40000, 41000, false);
+  const normalDuration = random(400, 500, false);
+
+  switch (transactionName) {
+    case alwaysSpikeTransactionName:
+      return spikedDuration;
+    case sometimesSpikeTransactionName:
+      return Math.random() > 0.01 ? spikedDuration : normalDuration;
```

Comment:
@dgieselaar I tried to produce a single spike but I was not able to. Synthtrace runs multiple instances in parallel so I  couldn't figure out how to just make one of them produce a spike and not all of them produce one. The end result was that I ended up with the same number of spikes as there was instances.

Reply:
add a third argument to `buildTransactions` called `shouldSpike` that you set to true on line 127 if in `.flatMap(serviceInstance, index)` index eq 0?

Reply:
Either I am misunderstanding what you mean, or you've forgotten how you architected synthtrace :D 
I am not talking about service instances but the actual nodejs instances spawned by synthtrace.

What you suggest would not work because every synthtrace instance runs a separate scenario where `i=0` in each of them.

Reply:
It seems like synthtrace runs with a single instance by default, but uses more instances for longer time ranges. Right now I am doing `--from="now-10h" --to="now"` which starts 9 synthtrace instances.

Reply:
oh you mean a Synthtrace instance. I don't understand why that is related. you're seemingly calculating a new duration for every event, with a 99-100% it is slow, and a 0-1% chance of it being fast (unless it is one of the other transaction names). Am I reading it incorrectly? What is a "spike" here?

Reply:
Disregard the current code. I'd like to be able run the scenario and produce exactly one spike (for `'GET /sometimes-spike'`)  at a random time

Reply:
A spike is a point where latency increases significantly.

Reply:
I would use the timestamp to decide how high the duration is. E.g. you check if the current time is in between 16:00 and 16:05 and then you calculate the diff between the boundaries of the time range and use some kind of math to calculate the avg duration. Let's say your normal duration is 100ms and you want to peak at 1000ms, over a 5m range, peaking right in the middle, you can do something like:

```ts
const duration = 100 + (isWithinPeakRange ? (2.5 - Math.abs( 2.5 - positionInTimeRange)) * 900 : 0);
```

(very much not tested)

Reply:
I'm not sure what you mean - each synthtrace instance gets a separate time slice so they don't know the total time. Is there a method I've overlooked to get the total time range? Perhaps I should add that?

Reply:
Either way, perhaps faster to resolve this on Monday if you around for a quick zoom sync. I also have some questions for this PR (specifically getting screen context to work for the insight component)

Reply:
You can add it to Synthtrace for sure, but you don't need to know the entire time range - just truncate it to a full day, and select a couple of minutes (or longer), and always generate data for at least 24 hours

Reply:
> and always generate data for at least 24 hours 

Hmm.. I don't control the time range. It's configurable by the consumer via the cli (`--from="now-10h" --to="now"`)

Reply:
I know, but I'm assuming you are running it 😄 .

Reply:
> I know, but I'm assuming you are running it 😄 .

Right now, yes, but the scenario can be run by anyone. I was hoping that regardless what time range they choose, that it will produce a single spike.

Reply:
(I haven't thought about `-live` mode - maybe it's a bad idea)